src/link/Wasm.zig

```diff
@@ -224,27 +239,15 @@ pub fn flushModule(self: *Wasm, comp: *Compilation) !void {
         const header_offset = try reserveVecSectionHeader(file);
         const writer = file.writer();
         for (self.funcs.items) |decl| {
-            const fn_data = &decl.fn_link.wasm.?;
-
-            // Write the already generated code to the file, inserting
-            // function indexes where required.
-            var current: u32 = 0;
-            for (fn_data.idx_refs.items) |idx_ref| {
-                try writer.writeAll(fn_data.code.items[current..idx_ref.offset]);
-                current = idx_ref.offset;
-                // Use a fixed width here to make calculating the code size
-                // in codegen.wasm.genCode() simpler.
-                var buf: [5]u8 = undefined;
-                leb.writeUnsignedFixed(5, &buf, self.getFuncidx(idx_ref.decl).?);
-                try writer.writeAll(&buf);
-            }
+            const fn_data = &decl.fn_link.wasm;
 
-            try writer.writeAll(fn_data.code.items[current..]);
+            // Write the already generated code to the file
+            try writer.writeAll(fn_data.code.items[decl.link.wasm.code_offset..]);
```

Comment:
How are incremental updates which shift function indexes handled without this logic?

Reply:
When allocateDeclIndexes is called, a check is done to see if the decl already exists in the `funcs` list. If not, they're appended and assigned an id. Meaning that if function indexes shift, they keep their old index. This same index id is then used for the function call. However, I just realized that this won't work for removing functions and that will mess up the order and therefore the id's. Thanks for your feedback @ifreund, I'll take another look at this tonight. Appreciate your time.

Reply:
Reverted to the old logic.